libs/screen/image.cpp

```diff
@@ -1159,8 +1159,191 @@ void _fillCircle(Image_ img, int cxy, int r, int c) {
     fillCircle(img, XX(cxy), YY(cxy), r, c);
 }
 
+typedef struct 
+{
+    int x, y;
+    int x0, y0;
+    int x1, y1;
+    int W,H;
+    int dx, dy;
+    int yi, xi;
+    int D;
+    int nextFuncIndex;
+}LineGenState;
```

Comment:
Here would be a good spot to add some comments

Reply:
```suggestion
}LineGenState; // For keep the scene state of generating Y values, will continue gen Y(s) with this state set, when we go next step with next X.
```